src/analysis/calculus/taylor.lean

```diff
@@ -0,0 +1,405 @@
+  rcases taylor_mean_remainder hx hf hf' gcont gdiff (λ _ _, by simp) with ⟨y, hy, h⟩,
+  use [y, hy],
+  rw h,
+  field_simp [nat.factorial_ne_zero n],
+  ring,
+end
+
+/-- **Taylor's theorem** with a uniform bound of the remainder
+
+We assume that `f` is `n+1`-times continuously differentiable on the closed set `Icc a b`.
+The difference of `f` and its `n`-th Taylor polynomial can be estimated by
+`C * (x - a)^(n+1) / n!` where `C` is a bound for the `n+1`-th iterated derivative of `f`. -/
+lemma taylor_mean_remainder_bound {f : ℝ → E} {a b C x: ℝ} {n : ℕ}
+  (hab : a ≤ b) (hf : cont_diff_on ℝ (n+1) f (Icc a b)) (hx : x ∈ Icc a b)
+  (hC : ∀ (y : ℝ) (hy : y ∈ Icc a b), ∥iterated_deriv_within (n + 1) f (Icc a b) y∥ ≤ C) :
+  ∥f x - taylor_within_eval f n (Icc a b) a x∥ ≤ C * (x - a)^(n+1) / n! :=
+begin
+  rcases eq_or_lt_of_le hab with rfl|h,
+  { rw [Icc_self, mem_singleton_iff] at hx,
+    simp [hx] },
+  -- The nth iterated derivative is differentiable
+  have hf' : differentiable_on ℝ (iterated_deriv_within n f (Icc a b)) (Icc a b) :=
+  hf.differentiable_on_iterated_deriv_within (with_top.coe_lt_coe.mpr n.lt_succ_self)
+    (unique_diff_on_Icc h),
+  -- natural numbers are non-negative
+  have fac_nonneg : 0 ≤ (n! : ℝ) := n!.cast_nonneg,
+  -- We can uniformly bound the derivative of the Taylor polynomial
+  have h' : ∀ (y : ℝ) (hy : y ∈ Ico a x),
+    ∥((n! : ℝ)⁻¹ * (x - y) ^ n) • iterated_deriv_within (n + 1) f (Icc a b) y∥
+    ≤ (n! : ℝ)⁻¹ * |(x - a)|^n * C, --has_Sup.Sup (g '' Icc a b),
+  { intros y hy,
+    rw [norm_smul, real.norm_eq_abs],
+    -- Estimate the iterated derivative by `Sup (g '' Icc a b)`
+    refine mul_le_mul _ (hC y ⟨hy.1, hy.2.le.trans hx.2⟩) (by positivity) (by positivity),
+    -- The rest is a trivial calculation
+    rw [abs_mul, abs_pow, abs_inv, nat.abs_cast],
+    mono*,
+    any_goals { positivity },
+    { exact hy.1 },
+    { linarith [hx.1, hy.2] } },
+  -- Apply the mean value theorem for vector valued functions:
+  have A : ∀ t ∈ Icc a x, has_deriv_within_at (λ y, taylor_within_eval f n (Icc a b) y x)
+    (((↑n!)⁻¹ * (x - t) ^ n) • iterated_deriv_within (n + 1) f (Icc a b) t) (Icc a x) t,
+  { assume t ht,
+    have I : Icc a x ⊆ Icc a b := Icc_subset_Icc_right hx.2,
+    exact (has_deriv_within_taylor_within_eval_at_Icc x h (I ht) hf.of_succ hf').mono I },
+  have := norm_image_sub_le_of_norm_deriv_le_segment' A h' x (right_mem_Icc.2 hx.1),
+  simp only [taylor_within_eval_self] at this,
+  refine le_trans this (le_of_eq _),
+  -- The rest is a trivial calculation
+  rw [abs_of_nonneg (sub_nonneg.mpr hx.1)],
+  ring_exp,
+end
+
+/-- **Taylor's theorem** with a uniform bound of the remainder
+
+We assume that `f` is `n+1`-times continuously differentiable on the closed set `Icc a b`.
+There exists a constant `C` such that for all `x ∈ Icc a b` The difference of `f` and its `n`-th
+Taylor polynomial can be estimated by `C * (x - a)^(n+1) / n!` where `C` is a bound for the `n+1`-th
```

Comment:
I don't see the point of the `1/n!` here. If you don't claim anything on the constant, can't you just remove it?